docs/thin_provision.md

```diff
@@ -0,0 +1,63 @@
+## Thin Provisioning
```

Comment:
let us add link to this doc in Readme, "Create a Storage class" section. Letting use know if they want to create the thin volume.

Reply:
[Storageclass.md ](https://github.com/openebs/lvm-localpv/blob/master/docs/storageclasses.md) already has the thin provision parameters info, So i just add the Thin Provision feature section in main README file